splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyCheckIT.java

```diff
@@ -706,6 +707,35 @@ public void multipleTablesReferencingSameTable() throws Exception {
     }
 
 
+    @Test
+    public void rowHistoryIsReadyCorrectly() throws Exception {
+            new TableCreator(conn)
+                    .withCreate("create table P(col1 int, col2 varchar(2), col3 int, col4 int, primary key (col2, col4))")
+                    .withInsert("insert into P values(?,?,?,?)")
+                    .withRows(rows(row(1, "a", 1, 1)))
+                    .create();
+            new TableCreator(conn)
+                    .withCreate("create table C (col1 int primary key, col2 varchar(2), col3 int, col4 int, constraint CHILD_FKEY foreign key(col2, col3) references P(col2, col4) )")
+                    .create();
+            conn.commit();
+
+        try(Connection c = newNoAutoCommitConnection()) {
+            c.setAutoCommit(false);
+
+            try (Statement statement = c.createStatement()) {
+                statement.executeUpdate("update P set col1 = 42");
+            }
+            c.rollback();
+        }
+
+        try(Connection conn = newNoAutoCommitConnection()) {
+            try(Statement s = conn.createStatement()) {
+                s.executeUpdate("insert into C values (400, 'a', 1, 42)"); // should not fail
+            }
+            conn.commit();
+        }
```

Comment:
```suggestion
        try(Connection c = newNoAutoCommitConnection();
        Connection c2 = newNoAutoCommitConnection();
        PreparedStatement ps = c2.preparedQuery("insert into C values (400, 'a', 1, 42)")) {
            c.setAutoCommit(false);

            try (Statement statement = c.createStatement()) {
                statement.executeUpdate("update P set col1 = 42");
            }
            c.rollback();
            
            ps.executeUpdate();
            c2.commit();
        }
```

Reply:
I think something like this would be better to reduce the window of time between the rollback and the update, but as it is it looks fine to me.